catalog/nato/stanag4559/catalog-stanag4559-common/src/main/java/alliance/catalog/nato/stanag4559/common/Stanag4559MetacardType.java

```diff
@@ -0,0 +1,644 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package alliance.catalog.nato.stanag4559.common;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.AttributeType;
+import ddf.catalog.data.MetacardType;
+import ddf.catalog.data.impl.AttributeDescriptorImpl;
+import ddf.catalog.data.impl.BasicTypes;
+
+public abstract class Stanag4559MetacardType implements MetacardType {
```

Comment:
I didn't extend from the Impl as I didn't want the code to have to construct and pass the list of attributedescriptors and name when setting the type. I did override the default hashCode and equals.